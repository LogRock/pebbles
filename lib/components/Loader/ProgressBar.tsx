import React from "react";
import styled from "styled-components";

export type ProgressBarProps = {
  percentage: number;
  size: "small" | "medium" | "large" | "xLarge";
};

const Bar = styled.div<Pick<ProgressBarProps, "size">>`
  position: relative;
  width: 100%;
  height: ${({ theme, size }) => theme.progressBar.sizes[size || "medium"]};
  overflow: hidden;

  border-radius: 48px;

  background-color: ${({ theme }) => theme.progressBar.bgColor};
`;

const InnerBar = styled.div`
  position: absolute;
  width: ${(props: ProgressBarProps) => `${props.percentage}%` || "0%"};
  height: ${({ theme, size }) => theme.progressBar.sizes[size || "medium"]};

  border-radius: 48px;

  background-color: ${({ theme }) => theme.progressBar.barColor};
`;

const ProgressBar = (props: ProgressBarProps) => {
  return (
    <Bar {...props}>
      <InnerBar {...props} />
    </Bar>
  );
};

export default ProgressBar;
