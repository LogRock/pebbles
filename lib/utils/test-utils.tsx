/* eslint-disable import/export */
import React from "react";
import { cleanup, render } from "@testing-library/react";
import { afterEach } from "vitest";
import { ThemeProvider } from "styled-components";
import { light as theme } from "../theme";

afterEach(() => {
  cleanup();
});

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
// override render export
export { customRender as render };
