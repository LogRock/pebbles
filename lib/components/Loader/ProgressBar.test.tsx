import React from "react";
import { render } from "../../utils/test-utils";
import ProgressBar from "./ProgressBar";

describe("ProgressBar tests", () => {
  it("matches the component snapshot", () => {
    const { container } = render(
      <ProgressBar percentage={75} size="medium"></ProgressBar>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
