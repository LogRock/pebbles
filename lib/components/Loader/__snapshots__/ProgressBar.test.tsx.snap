// Vitest Snapshot v1

exports[`ProgressBar tests > shows the number percentage 1`] = `
<div
  class="sc-bdvvtL gmMwPz"
>
  <div
    class="sc-gsDKAQ drCyrQ"
  />
</div>
`;
